openpype/pipeline/colorspace.py

```diff
@@ -118,6 +169,20 @@ def get_imageio_colorspace_from_filepath(
         if ext_match and file_match:
             colorspace_name = file_rule["colorspace"]
 
+    # if no file rule matched, try to get colorspace
+    # from filepath with OCIO v2 way
+    if (
+        compatibility_check_config_version(config_data["path"], major=2)
+        and not colorspace_name
+    ):
+        colorspace_name = get_colorspace_from_filepath(
+            config_data["path"], path)
+
+    # use parse colorspace from filepath as fallback
+    colorspace_name = colorspace_name or parse_colorspace_from_filepath(
+        path, config_path=config_data["path"]
+    )
```

Comment:
By adding these into this function the docstring and function names feels incorrect - this isn't at all about imageio file rules in settings suddenly but it's doing way more fallbacks than that. Even worse is that with this here it's now even more unclear what the difference is between `get_colorspace_from_filepath`, `parse_colorspace_from_filepath` and `get_imageio_colorspace_from_filepath` since they start overlapping so much. I feel like adding these here is the wrong approach. If we want a "match any of the rules" function we'll likely just need a dedicated function for that which then combines all three in the right order.

With this added here it's now also to JUST compare `imageio filerules` with the filepath.

Reply:
Agree with you on this. Look to a6b8fdfe33dc5dcf1eb1120a65f5ad264b3b1ef4 if it makes more sence this way.